### DISCUSSION TITLE
Porting to MOD DWARF possible

### DISCUSSION THREAD

**Discussion by spunktsch** [Dougal-s/Aether#17] [Q&A] [answered]
Hi, 

I'm really fond of your lv2 port of this reverb. I'd also would like to use it with my MOD DWARF - which is a Linux based MultiFX Pedal.
Is it possible to build it for the platform?

If you want to know more  check out https://moddevices.com/platform/mod-os/ and https://wiki.moddevices.com/wiki/How_To_Build_and_Deploy_LV2_Plugin_to_MOD_Duo. The Github is https://github.com/moddevices

Cheers 
Sebastian

**Comment by Dougal-s** [comment, marked as answer]
Sorry for the slow reply.
Aether now compiles for the MOD DWARF! Unfortunately, I have no way of verifying whether the plugin works correctly, so I would be grateful if you could check if the attached lv2 bundle works. I've also attached the buildroot .mk in case you want to compile it yourself using [MOD Plugin Builder](https://github.com/moddevices/mod-plugin-builder).

[aether-aarch64-mod.zip](https://github.com/Dougal-s/Aether/files/7954674/aether-aarch64-mod.zip)
[aether.mk.zip](https://github.com/Dougal-s/Aether/files/7947346/aether.mk.zip)

**Reply by spunktsch** [reply]
thank you very much. I've tested it with lv2 bundle on the dwarf and it works great. 
As expected it is very CPU intensive - so I have to limit the delay lines and a few other things. But still sounds great. 

I'll point some users of the [MOD Forum](https://forum.moddevices.com/) to your github - someone else might be interested.

And maybe you want to consider putting it through the MODTeam and upload it to their Pedalshop. 

Again, thanks for making this available and dm me if you consider putting it into the MODShop and need help with some graphics.

**Reply by malfunction54** [reply]
I got the aether plugin built and deployed on my MOD Dwarf and I think it sounds great.  Are you interested in getting it into the plugin store?  I'd be happy to help with that process as well. (not a MOD employee, just an enthusiast).  One question, though: are there some compile options that might reduce the CPU usage a bit?  Thanks!

**Reply by Dougal-s** [reply]
Sorry for taking so long to reply. I've been/still am busy.

Getting Aether onto the plugin store is definitely something I am interested in! There are several things to do first, listed on this [page](https://wiki.moddevices.com/wiki/Plugin_Criteria) on the mod devices wiki, but the primary tasks are

1. reducing cpu usage and
2. creating a mod ui

I'll probably need the most help with no. 2 since I'm unfamiliar with the mod devices ecosystem.

As for compilation flags, there aren't any that I'm aware of. Release already uses `-Ofast`, and it seems like the MOD Plugin Builder toolchain already optimizes for the target mod device. At least that's the impression I got from looking at the result of passing `--target-help`.

**Reply by jjYBdx4IL** [reply]
What's the CPU usage issue? UI related? Be aware that there still seems to be an issue with nvidia+opengl: https://web.archive.org/web/20120710044621/http://forum.openscenegraph.org/viewtopic.php?t=3653#18283 ie. if on Windows you have one core at 100%, this is likely the cause. But the MSVC profiler will tell you that.

**Reply by Dougal-s** [reply]
Huh, I wasn't aware of that! In this case, the cpu usage is dsp related since Aether is compiled without a UI for mod devices.

**Reply by malfunction54** [reply]
I have been using the plugin on a MOD Dwarf and I found that reducing the late delay lines to "1" significantly reduces CPU usage.  What is interesting is that that having the plugin by itself consumes about 40% CPU, but adding it to a pedalboard that has other plugins already only increases CPU by a few percent, usually less than +10%.  So for typical pedalboards, it is not an issue.  Results may vary on the Duo and DuoX.

**Reply by malfunction54** [reply]
I submitted your .mk file to @falkTX and he pushed Aether to the beta store.  There are a few people on the forums who have contributed UI assets to other plugins.  I bet they'd be willing to help out in that area :)

**Reply by spunktsch** [reply]
@Dougal-s great to hear. 
I could definitely help with the gui (mostly css and svg). I would suggest to stick closely to your original design.
Also I highly recommend you to post in the mod forum. We started a thread [there ](https://forum.moddevices.com/t/aether-reverb-cloudseed/7273/22) and i'm sure there are a few people that would help.
